apis/Google.Storage.V1/Google.Storage.V1/UrlSigner.cs

```diff
@@ -36,14 +36,45 @@ namespace Google.Storage.V1
     /// <remarks>
     /// See https://cloud.google.com/storage/docs/access-control/signed-urls for more information on signed URLs.
     /// </remarks>
-    public static class SignedUrlUtils
+    public sealed class UrlSigner
     {
         private const string GoogHeaderPrefix = "x-goog-";
         private const string GoogEncryptionKeyHeader = "x-goog-encryption-key";
         private const string GoogEncryptionKeySHA256Header = "x-goog-encryption-key-sha256";
         private const string StorageHost = "https://storage.googleapis.com";
         private static readonly DateTimeOffset UnixEpoch = new DateTimeOffset(new DateTime(1970, 1, 1, 0, 0, 0, DateTimeKind.Utc), TimeSpan.Zero);
 
+        private readonly ServiceAccountCredential.Initializer _credentials;
+
+        /// <summary>
```

Comment:
I'd be tempted to use a private constructor and static factory methods:

- FromServiceAccountPath
- FromServiceAccountData

(or something similar)

That way if we want something else that takes a string - e.g. an overload of `FromServiceAccountData` with the JSON text itself - we can do so easily. At that point, one of these can easily call the other :)